tests/test_transforms/test_decomposition/test_dft_based.py

```diff
@@ -0,0 +1,344 @@
+from etna.transforms import FourierDecomposeTransform
+from etna.transforms import IForestOutlierTransform
+from etna.transforms import TimeSeriesImputerTransform
+
+
+def simple_pipeline_with_decompose(in_column, horizon, k):
+    pipeline = Pipeline(
+        transforms=[FourierDecomposeTransform(k=k, in_column=in_column)],
+        model=HoltWintersModel(),
+        horizon=horizon,
+    )
+    return pipeline
+
+
+@pytest.fixture()
+def ts_with_exogs() -> TSDataset:
+    periods = 100
+    periods_exog = periods + 10
+    df = generate_ar_df(start_time="2020-01-01", periods=periods, freq="D", n_segments=2)
+    df_exog = generate_ar_df(start_time="2020-01-01", periods=periods_exog, freq="D", n_segments=2, random_seed=2)
+    df_exog.rename(columns={"target": "exog"}, inplace=True)
+    df_exog["holiday"] = np.random.choice([0, 1], size=periods_exog * 2)
+
+    df = TSDataset.to_dataset(df)
+    df_exog = TSDataset.to_dataset(df_exog)
+    ts = TSDataset(df, freq="D", df_exog=df_exog, known_future="all")
+    return ts
+
+
+@pytest.fixture()
+def ts_with_exogs_train_test(ts_with_exogs):
+    return ts_with_exogs.train_test_split(test_size=20)
+
+
+@pytest.fixture()
+def forward_stride_datasets(ts_with_exogs):
+    train_df = ts_with_exogs.df.iloc[:-10]
+    test_df = ts_with_exogs.df.iloc[-20:]
+
+    train_ts = TSDataset(df=train_df, freq=ts_with_exogs.freq)
+    test_ts = TSDataset(df=test_df, freq=ts_with_exogs.freq)
+
+    return train_ts, test_ts
+
+
+@pytest.fixture()
+def ts_with_missing(ts_with_exogs):
+    target_df = ts_with_exogs[..., "target"]
+    target_df.iloc[10] = np.nan
+
+    return TSDataset(df=target_df, freq=ts_with_exogs.freq)
+
+
+@pytest.mark.parametrize("in_column", ("target", "feat"))
+def test_init(in_column):
+    transform = FourierDecomposeTransform(k=5, in_column=in_column)
+    assert transform.required_features == [in_column]
+    assert transform._first_timestamp is None
+    assert transform._last_timestamp is None
+
+
+@pytest.mark.parametrize("k", (-1, 0))
+def test_invalid_k(k):
+    with pytest.raises(ValueError, match="Parameter `k` must be positive integer!"):
+        FourierDecomposeTransform(k=k, in_column="target")
+
+
+@pytest.mark.parametrize(
+    "series, answ",
+    (
+        (pd.Series([1]), 1),
+        (pd.Series([1, 2]), 2),
+        (pd.Series([1, 2, 3]), 2),
+        (pd.Series([1, 2, 3, 4]), 3),
+        (pd.Series([1, 2, 3, 4, 5]), 3),
+        (pd.Series([1, 2, 3, 4, 5, 6]), 4),
+    ),
+)
+def test_get_num_pos_freqs(series, answ):
+    res = FourierDecomposeTransform._get_num_pos_freqs(series=series)
+    assert res == answ
+
+
+def test_check_segments_missing_values(ts_with_missing):
+    df = ts_with_missing[..., "target"]
+    transform = FourierDecomposeTransform(k=5)
+    with pytest.raises(ValueError, match=f"Feature `target` contains missing values"):
+        transform._check_segments(df=df)
+
+
+def test_check_segments_large_k(ts_with_exogs):
```

Comment:
Now we are setting `k=100`, could we also (using parametrization) set some minimal value that leads to error on `ts_with_exogs`? It could better detect test regression in the future.

Reply:
Added test